src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,404 @@
+﻿using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   Tweaked colour picker allows for tooltip texts translations and has a customized theming.
+/// </summary>
+public class TweakedColourPicker : ColorPicker
+{
+    /// <summary>
+    ///   This is where presets are stored after a colour picker exited scene tree.
+    ///   <remarks>
+    ///     The Key is the group name; <br />
+    ///     The Value is the preset storage.
+    ///   </remarks>
+    /// </summary>
+    private static readonly Dictionary<string, PresetGroupStorage> PresetsStorage
+        = new Dictionary<string, PresetGroupStorage>();
+
+    private readonly List<TweakedColourPickerPreset> presets = new List<TweakedColourPickerPreset>();
+
+    private HSlider sliderROrH;
+    private HSlider sliderGOrS;
+    private HSlider sliderBOrV;
+    private HSlider sliderA;
+    private ToolButton pickerButton;
+    private CheckButton hsvCheckButton;
+    private CheckButton rawCheckButton;
+    private LineEdit htmlColourEdit;
+    private HSeparator separator;
+    private GridContainer presetsContainer;
+    private TextureButton addPresetButton;
+
+    private bool hsvButtonEnabled = true;
+    private bool rawButtonEnabled = true;
+    private bool presetsEnabled = true;
+    private bool presetsVisible = true;
+
+    private PresetGroupStorage groupStorage;
+
+    private delegate void AddPresetDelegate(Color colour);
+
+    private delegate void DeletePresetDelegate(Color colour);
+
+    [Export]
+    public string PresetGroup { get; private set; } = "default";
+
+    /// <summary>
+    ///   Decide if user can toggle HSV CheckButton to switch HSV mode.
+    /// </summary>
+    [Export]
+    public bool HSVButtonEnabled
+    {
+        get => hsvButtonEnabled;
+        set
+        {
+            hsvButtonEnabled = value;
+            UpdateButtonsState();
+        }
+    }
+
+    /// <summary>
+    ///   Decide if user can toggle Raw CheckButton to switch Raw mode.
+    /// </summary>
+    [Export]
+    public bool RawButtonEnabled
+    {
+        get => rawButtonEnabled;
+        set
+        {
+            rawButtonEnabled = value;
+            UpdateButtonsState();
+        }
+    }
+
+    /// <summary>
+    ///   Change the picker's HSV mode.
+    ///   <remarks>
+    ///     This is named not HSVMode because this hides a Godot property to avoid break custom functions.
```

Comment:
```suggestion
    ///     This is not named HSVMode because this hides a Godot property to avoid breaking custom functions.
```
Even with that grammar fix I'm still not sure what "avoid breaking custom functions" means here?

Reply:
Well, we have custom buttons. If directly access Godot HsvMode, custom buttons will not be updated.

Reply:
Could you try to explain the problem in the form of a comment?